lib_eio_luv/eio_luv.ml

```diff
@@ -228,6 +248,23 @@ module File = struct
     await_with_cancel ~request (fun loop -> Luv.File.mkdir ~loop ~request ~mode path)
 end
 
+module Poll = struct
+
+  let poll_readable fd callback =
+    let p = Luv.Poll.init ~loop:(get_loop ()) (Obj.magic fd) in
+    match p with
+      | Ok poll ->
+        await_with_cancel_poll ~poll (fun _loop _fibre -> Luv.Poll.start poll [`READABLE;] (fun _ -> callback ()))
```

Comment:
You can't throw away the calling fibre here. It needs to be resumed with the result.